apps/omg_watcher/lib/exit_processor/core.ex

```diff
@@ -1274,13 +1274,27 @@ defmodule OMG.Watcher.ExitProcessor.Core do
    - a block number which can be used to retrieve needed information to challenge or an ife which spends inputs
    - the relevant exit information
   """
-  @spec get_challenge_data(tuple(), Utxo.Position.t(), t()) ::
+  @spec get_challenge_data(tuple(), Utxo.Position.t(), Block.t() | :not_found, t()) ::
           {:ok, pos_integer() | KnownTx.t(), ExitInfo.t()} | {:error, atom()}
-  def get_challenge_data(spending_blknum_response, exiting_utxo_pos, %__MODULE__{exits: exits} = state) do
+  def get_challenge_data(spending_blknum_response, exiting_utxo_pos, block, %__MODULE__{exits: exits} = state) do
     with %ExitInfo{} = exit_info <- Map.get(exits, exiting_utxo_pos, {:error, :exit_not_found}),
+         exit_txhash <- get_standard_exit_txhash(exit_info, exiting_utxo_pos, block),
          ife_response = get_ife_based_on_utxo(exiting_utxo_pos, state),
          {:ok, raw_spending_proof} <- ensure_challengeable(spending_blknum_response, ife_response),
-         do: {:ok, raw_spending_proof, exit_info}
+         do: {:ok, raw_spending_proof, %{exit_info | exit_txhash: exit_txhash}}
+  end
+
+  defp get_standard_exit_txhash(
```

Comment:
comment both clauses of `get_standard_exit_txhash`

1/ `# get the hash of a deposit transaction`
2/ `# get the hash of a transaction from a block`